.github/workflows/github-dev-snap.yaml

```diff
@@ -0,0 +1,129 @@
+# Copyright 2023 Pluto TV
+# Copyright 2022 Google LLC
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+name: GitHub Development Snapshot Build
+
+on:
+  push:
+    branches: [ feature/TRANS-4831-nighty-builds ]
+  workflow_dispatch:
+
+env:
+  DEVELOPMENT_BRANCH: "pluto-cmake"
+  # default development tag
+  # TODO(sasha): provide tag on workflow dispatch
+  DEVELOPMENT_TAG: "develop"
+  ARTIFACT_NAME: "develop"
+
+jobs:
+  build_linux:
+    name: Build artifacts and upload to GitHub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install build tools
+        run: |
+          sudo apt update && sudo apt install -y \
+              cmake
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.DEVELOPMENT_BRANCH }}
+          submodules: recursive
+
+      - name: Apply ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+
+      - name: Generate build files
+        run: |
+          export PATH="/usr/lib/ccache:/usr/local/opt/ccache/libexec:$PATH"
+
+          mkdir -p build/
+
+          cmake \
+            -DCMAKE_BUILD_TYPE="Release" \
+            -DBUILD_SHARED_LIBS=1 \
+            -S . \
+            -B build/
+
+      - name: Build
+        run: cmake --build build/
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.ARTIFACT_NAME }}
+          path: build/packager/libpackager.so
```

Comment:
Could we also add a step to upload the `packager` directory along with the dynamic library so all the header files can be bundled with the so file?  It looks like this action supports directories too: https://github.com/actions/upload-artifact